client/app/components/empty-state/EmptyState.jsx

```diff
@@ -17,131 +17,164 @@ function createDashboard() {
+  urlText: null,
+  onClick: null,
+};
+
 export function EmptyState({
   icon,
   title,
   description,
   illustration,
   helpLink,
   onboardingMode,
-  ...showSteps
+  showAlertStep,
+  showDashboardStep,
+  showInviteStep,
 }) {
-  const steps = filter(getSteps(showSteps), step => step.show);
-  const shouldShow = !onboardingMode || some(steps, { completed: false });
-
-  if (shouldShow) {
-    return (
-      <div className="empty-state bg-white tiled">
-        <div className="empty-state__summary">
-          {title && <h4>{title}</h4>}
-          {icon && <h2><i className={icon} /></h2>}
-          <p>{description}</p>
-          <img
-            src={'/static/images/illustrations/' + illustration + '.svg'}
-            alt={illustration + 'Illustration'}
-            width="75%"
-          />
-        </div>
-        <div className="empty-state__steps">
-          <h4>Let&apos;s get started</h4>
-          <ol>
-            {map(steps, (step, key) => (
-              <li key={key} className={classNames({ done: step.completed })}>{step.render(step)}</li>
-            ))}
-          </ol>
-          <p>
-            Need more support?{' '}
-            <a href={helpLink} target="_blank" rel="noopener noreferrer">
-              See our Help
-              <i className="fa fa-external-link m-l-5" aria-hidden="true" />
-            </a>
-          </p>
-        </div>
-      </div>
-    );
+  const isAvailable = {
+    dataSource: true,
+    query: true,
+    alert: showAlertStep,
+    dashboard: showDashboardStep,
+    inviteUsers: showInviteStep,
+  };
+
+  const isCompleted = {
+    dataSource: organizationStatus.objectCounters.data_sources > 0,
+    query: organizationStatus.objectCounters.queries > 0,
+    alert: organizationStatus.objectCounters.alerts > 0,
+    dashboard: organizationStatus.objectCounters.dashboards > 0,
+    inviteUsers: organizationStatus.objectCounters.users > 1,
+  };
+
+  // Show if `onboardingMode=false` or any requested step not completed
+  const shouldShow = !onboardingMode || some(keys(isAvailable), step => isAvailable[step] && !isCompleted[step]);
+
+  if (!shouldShow) {
+    return null;
   }
 
-  return null;
+  return (
+    <div className="empty-state bg-white tiled">
+      <div className="empty-state__summary">
+        {title && <h4>{title}</h4>}
+        <h2>
+          <i className={icon} />
+        </h2>
+        <p>{description}</p>
+        <img
+          src={'/static/images/illustrations/' + illustration + '.svg'}
+          alt={illustration + ' Illustration'}
+          width="75%"
+        />
+      </div>
+      <div className="empty-state__steps">
+        <h4>Let&apos;s get started</h4>
+        <ol>
+          {currentUser.isAdmin && (
+            <Step
+              show={isAvailable.dataSource}
+              completed={isCompleted.dataSource}
+              url="data_sources/new"
+              urlText="Connect"
+              text="a Data Source"
+            />
```

Comment:
That's an example of what I mentioned earlier: in Angular template we'd put everything in a single line :smile: